File: backend/src/main/java/protasks/backend/Rol/Priority.java
package protasks.backend.Rol;

import java.awt.*;

public enum Priority {
    // Color int shows the equivalence in android Color library
    NO_PRIORITY("No Priority", -1, 0),
    LOW("Low", -7829368, 1),//Color Gray
    NORMAL("Normal", -16711936, 2),//Color Green
    HIGH("High", -256, 3), //Color Yellow
    URGENT("Urgent", -65536, 4); //Color Red

    public int getIndex() {
        return this.index;
    }

    // custom properties with default values
    private final String printableName;
    private final Integer color;
    private final Integer index;

    Priority(String printableName, Integer color, Integer index) {
        this.printableName = printableName;
        this.color = color;
        this.index = index;
    }

    public int[] getColors() {
        Priority[] values = values();
        int[] colors = new int[values.length];
        for (int i = 0; i < colors.length; i++) {
            colors[i] = values[i].color;
        }
        return colors;
    }

    public String[] getNames() {
        Priority[] values = values();
        String[] names = new String[values.length];
        for (int i = 0; i < names.length; i++) {
            names[i] = values[i].printableName;
        }
        return names;
    }


}

File: app/src/main/java/com/example/protasks/models/Priority.kt
package com.example.protasks.models

import android.graphics.Color
import java.lang.Exception


enum class Priority {
    NO_PRIORITY("No Priority",-1,0),
    LOW("Low",Color.GRAY,1),
    NORMAL("Normal",Color.GREEN,2),
    HIGH("High",Color.YELLOW,3),
    URGENT("Urgent",Color.RED,4);

    fun getIndex(): Int {
        return this.index!!
    }

    // custom properties with default values
    private var printableName : String? = null
    private var color: Int? = null
    private var index: Int? =null

    constructor()

    // custom constructors
    constructor(
        printableName: String,
        color: Int,
        index:Int
    ) {
        this.color = color
        this.printableName = printableName
        this.index=index
    }

    companion object{
        fun getColors():IntArray{
            val values = values()
            val colors = IntArray(values.size)
            values.forEachIndexed { index, element ->
                colors[index] = element.color!!
            }
            return colors
        }

        fun getNames():Array<String>{
            val values = values()
            val names=Array(values().size) { "" }
            values.forEachIndexed { index, element ->
                names[index] = element.printableName!!
            }
            return names
        }
        fun getPriorityByPrintableName(printableName: String):Priority{
            for (p in values()){
                if (p.printableName==printableName){
                    return p
                }
            }
            throw Exception("There is no Priority with this printable name")
        }

    }

}

